### PR TITLE
[feature-wip](parquet-reader) parquert scanner can read data

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

Issue Number: close #xxx

## Problem summary

finished first version of the new parquet scanner and reader, this version just support primary type

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/file_hdfs_scanner.cpp [hunk 1]
```diff
@@ -67,6 +67,7 @@ Status ParquetFileHdfsScanner::_get_next_reader(int _next_range) {
             _reader->init_reader(tuple_desc, _file_slot_descs, _conjunct_ctxs, _state->timezone());
     if (!status.ok()) {
         if (status.is_end_of_file()) {
+            LOG(WARNING) << "is_end_of_file: " << status.is_end_of_file();
```
Comment (cambyzju) [inline be/src/vec/exec/file_hdfs_scanner.cpp:70]: it is better not print warning log for logic error.

**Comment by cambyzju** [comment]
please add some description about what this pr doing.

**Comment by wsjz** [comment]
> please add some description about what this pr doing.

use new parquet reader instead arrow reader WIP

File: be/src/vec/exec/file_hdfs_scanner.cpp [hunk 2]
```diff
@@ -28,6 +28,10 @@ ParquetFileHdfsScanner::ParquetFileHdfsScanner(RuntimeState* state, RuntimeProfi
                                                ScannerCounter* counter)
         : HdfsFileScanner(state, profile, params, ranges, pre_filter_texprs, counter) {}
 
+ParquetFileHdfsScanner::~ParquetFileHdfsScanner() {
+    ParquetFileHdfsScanner::close();
```
Comment (Gabriel39) [inline be/src/vec/exec/file_hdfs_scanner.cpp:32]: I think we should move this call to `FileScanNode` when this `FileScanner` is ready to close.

**Comment by Gabriel39** [comment]
Hi @wsjz I have fixed a failure regression case and you need rebase master and push again to resolve this.

**Comment by wsjz** [comment]
> Hi @wsjz I have fixed a failure regression case and you need rebase master and push again to resolve this.

OK

File: be/src/vec/exec/file_hdfs_scanner.cpp (hunk 2 above)
Comment (wsjz) [reply be/src/vec/exec/file_hdfs_scanner.cpp:32]: I think FileScanNode in this version we can keep the same with before

**Review by morningman — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**